tests/playwright/src/model/pages/extension-details-page.ts

```diff
@@ -23,22 +23,32 @@ import { BasePage } from './base-page';
 import { ExtensionsPage } from './extensions-page';
 
 export class ExtensionDetailsPage extends BasePage {
+  readonly header: Locator;
+  readonly tabs: Locator;
+  readonly tabContent: Locator;
+  readonly buttonGroup: Locator;
```

Comment:
I don't really see why these are needed, it looks like needless abstraction and complexity where none if required, what problem do these container objects are meant to solve?

Reply:
This is the representation of the dom structure on the page.

Reply:
I know that, but they are superfluous to the goal, which is to handle the webelement, there is no benefit in getting the buttonGroup before getting the actual buttons for example, it adds an extra abstraction layer that has no benefit but increases complexity.

Just because a container is displayed in the DOM doesn't mean it needs to be present in the POM model, POM models should be limited to only what is useful, it should not have superfluous elements added to it.

Reply:
These containers have their reason, it is aria functionality, why not to include them if we have them in the structure? I understand that it is adding more complexity in this case, but if someone decides to remove the reasonable grouping of the actions on the page, we would know it, since tests will start to fail. Additionally, any changes to the page (like additional actions added for any reason) wont break anything. Also one does not really goes into class and study the particular containers and  their alignment, he uses locators he searches for. 
So I still think this has its value.

Reply:
The containers have their reason from a development perspective, but not from a testing/functional perspective, just because they are there to help the devs frame something doesn't mean that it brings value to testing the end product. The goal of e2e testing is to test functionality, not some abstract concepts.

e2e testing shouldn't care of some container element that frames the buttons changes from X to Y, the tests shouldnt break if no functionality is broken, but that is exactly what you are introducing with this abstractions, because now if the container object changes or gets renamed the test will fail, which is wrong because the functionality is the same and it works.

You are confusing module/unit tests with e2e tests and are trying to squeeze both of them at the same time, this is the wrong approach, these containers should not be in scope for e2e tests.

Reply:
I am not confusing component tests with e2e test. we need to have robust e2e test that wont break if something new is added. It is done in images, containers pages, etc. So I wont change that concept since it is right. How is it aligned inside the hierarchy is not visible during usage. I am fine with this approach. If we wont agree, we might ask others since we have more people that are affected.

Reply:
> we need to have robust e2e test that wont break if something new is added

But this approach is exactly the opposite of this goal, having robust tests is good and this manner of abstraction leads to more flakyness because we end up with e2e tests that fail because somebody change the name of some container that holds some buttons somewhere from X to Y, while the buttons are operating completely fine.

>I am fine with this approach

But why are you fine with it when it adds complexity and flakyness and brings no benefit to testing functionality of the app?

Reply:
We can take a look at opposite situation, someone adds another view into page (ie. for dependency that extension now can have). With precise usage of container, tests wont be affected. They will still be testing the same functionality. Without it, it is going to fail. 
So I am fine adding complexity, since only complexity is the one during pr review. Making it harder to read, I agree. On the other hand, I disagree with flakiness. It either fails or pass. 
Tests should fail is something is changed, ideally in before the code is merged. That counts for both cases.

Reply:
I am finding it much harder to think of not adding more abstraction next time to just avoid long discussion under PR review. For me, we should respect the structure as it is (any extreme is bad), and not to overthink just not to add more properties into poms. @ScrewTSW @amisskii @xbabalov any opinion would be nice! I might be thinking about it wrong.

Reply:
>Without it, it is going to fail.

No, that's simply not true, if your test fails from such a thing it means whatever locators you are using are not sufficient and/or wrong, and that means when you fix it you make it better and more robust, on the other hand the way you are doing it will always be vulnerable to container name changes or simply changing the container.

On top of that yes it absolutely adds complexity to code review but it also adds complexity to test debugging and to test writing as can be seen literally in this PR.

Reply:
So it is true then, test is gonna fail, and you need to fix it (locators), which is alright. If I do not add it, it might also fail. I do not see to much reasoning between these two. Both situation requires update to the pom. One will fail for not finding the locator, the other for ambiguity. 

It has zero impact on test writing, you do not use directly the parts in between, you are using the end elements.

But it is true that debugging of more elements in the failed locator assertion is harder. Alright then, will remove it.